tests/integration/v3_watch_restore_test.go

```diff
@@ -81,14 +87,30 @@ func TestV3WatchRestoreSnapshotUnsync(t *testing.T) {
 	}
 
 	clus.Members[0].InjectPartition(t, clus.Members[1:]...)
-	initialLead := clus.WaitMembersForLeader(t, clus.Members[1:])
+	initialLead := clus.WaitMembersForLeader(t, clus.Members[1:]) + 1
 	t.Logf("elected lead: %v", clus.Members[initialLead].Server.MemberId())
 	t.Logf("sleeping for 2 seconds")
 	time.Sleep(2 * time.Second)
 	t.Logf("sleeping for 2 seconds DONE")
 
 	kvc := integration.ToGRPC(clus.Client(1)).KV
 
+	// NOTE: In 3 members cluster, after initial lead has been elected,
+	// there are 3xConfChange + 3xMemberAttrSet + 1xClusterVersionSet logs.
+	// So member 0 has index 8 in raft log before network partition. We
+	// need to trigger snapshot at least twice.
```

Comment:
nit: `need to trigger EtcdServer.snapshot()` 
just a clarification for anyone new to this.

Reply:
Updated!